backend/de.metas.serviceprovider/src/main/java/de/metas/serviceprovider/external/label/IssueLabelRepository.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * #%L
+ * de.metas.serviceprovider.base
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.serviceprovider.external.label;
+
+import com.google.common.collect.ImmutableList;
+import de.metas.organization.OrgId;
+import de.metas.serviceprovider.issue.IssueId;
+import de.metas.serviceprovider.model.I_S_IssueLabel;
+import de.metas.util.Check;
+import lombok.NonNull;
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Repository
+public class IssueLabelRepository
+{
+	private final IQueryBL queryBL;
+
+	public IssueLabelRepository(final IQueryBL queryBL)
+	{
+		this.queryBL = queryBL;
+	}
+
+	public ImmutableList<IssueLabel> getByIssueId(@NonNull final IssueId issueId)
+	{
+		return getRecordsByIssueId(issueId)
+				.stream()
+				.map(this::of)
+				.collect(ImmutableList.toImmutableList());
+	}
+
+	public void persistLabels(@NonNull final IssueId issueId,
+							  @NonNull final ImmutableList<IssueLabel> issueLabels)
+	{
+		final ImmutableList<I_S_IssueLabel> newLabels =
+				issueLabels
+						.stream()
+						.map(label -> of(issueId, label))
+						.collect(ImmutableList.toImmutableList());
+
+		final ImmutableList<I_S_IssueLabel> existingLabels = getRecordsByIssueId(issueId);
+
+		persistLabels(newLabels, existingLabels);
+	}
+
+	@NonNull
+	private ImmutableList<I_S_IssueLabel> getRecordsByIssueId(@NonNull final IssueId issueId)
+	{
+		return queryBL.createQueryBuilder(I_S_IssueLabel.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_S_IssueLabel.COLUMNNAME_S_Issue_ID, issueId.getRepoId())
```

Comment:
do you have an index on `S_IssueLabel.S_Issue_ID` ?

Reply:
yep